### PR TITLE
Upgrade superagent to 8.1.2 & typescript to 5.2.2; Point integration-tests to mountebank@2.9.1

### PR DESCRIPTION
**Pull request by yeongsheng-tan**
- Upgrade `superagent` to 8.1.2
- Upgrade `typescript` to 5.2.2
- Upgrade `tsconfig.json` relevant to `typescript@5.2.2`
- Upgrade `eslint`, `typescript-eslint`, `mocha`, `chai`, `prettier` to latest relevant versions
- change `integration-test` to run against `mountebank@2.9.1`

### CODE REVIEW

**Comment by yeongsheng-tan** [comment]
@AngelaE would you happen to have sometime to help review and merge this PR please. Thank you.

File: integration-tests/package.json [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
     "pretty": "prettier -w ./**/*",
     "lint": "eslint . --ext .ts",
     "lint-fix": "eslint . --ext .ts --fix",
-    "mb:up": "docker run -d --name mb -p 2525:2525 -p 12345:12345 -p 12346:12346 bbyars/mountebank:2.6.0 mb start",
-    "mb:up-on-port-2524": "docker run -d --name mb -p 2524:2525 -p 12345:12345 -p 12346:12346 bbyars/mountebank:2.6.0 mb start",
+    "mb:up": "docker run -d --name mb -p 2525:2525 -p 12345:12345 -p 12346:12346 bbyars/mountebank:2.9.1 mb start",
+    "mb:up-on-port-2524": "docker run -d --name mb -p 2524:2525 -p 12345:12345 -p 12346:12346 bbyars/mountebank:2.9.1 mb start",
     "test-different-mb-url": "MB_PORT=2524 npm run test-ci -- -g ^Mountebank",
-    "mb:down": "docker kill mb && docker rm mb"
+    "mb:down": "docker kill mb && podman rm mb"
```
Comment (AngelaE) [inline integration-tests/package.json:15]:
```suggestion
    "mb:down": "docker kill mb && docker rm mb"
```

Did this get changed accidentally?

File: integration-tests/package.json [hunk 2]
```diff
@@ -3,16 +3,16 @@
   "version": "1.0.0",
   "description": "Integration tests for @anev/ts-mountebank ",
   "scripts": {
-    "test": "mocha --recursive ./lib",
+    "test": "mocha --recursive ./dist",
```
Comment (AngelaE) [inline integration-tests/package.json:6]: 😍 Thanks for spotting!

**Review by AngelaE — COMMENTED** [review]
Thanks for the upgrade! Including the pipelines :)
And apologies for responding so late!
Looks great, apart from one minor comment around podman. Any chance you could revert that?

I'll aim to merge and release the new package this weekend.

File: integration-tests/package.json (hunk 1 above)
Comment (yeongsheng-tan) [reply integration-tests/package.json]: Fixed. Sorry for the noob mistake.

**Comment by yeongsheng-tan** [comment]
@AngelaE  Thanks for the PR merge. Wondering if you plan to cut a new version and deploy to [npmjs](https://www.npmjs.com/)?

**Comment by AngelaE** [comment]
> @AngelaE Thanks for the PR merge. Wondering if you plan to cut a new version and deploy to [npmjs](https://www.npmjs.com/)?

Looking at that right now. If I run into problems today (am on a new computer) I'll do it tomorrow.

**Comment by AngelaE** [comment]
This is published @yeongsheng-tan please let me know if there are problems.

**Comment by yeongsheng-tan** [comment]
Awesome. Thanks. I can start to consume the new version now. Cheers and very much appreciate your prompt attention! 👍

**Pull request merged**